apps/server/src/modules/tool/tool.module.ts

```diff
@@ -1,18 +1,40 @@
 import { Module } from '@nestjs/common';
 import { LtiToolRepo, PseudonymsRepo } from '@shared/repo';
-import { Lti11ResponseMapper } from '@src/modules/tool/mapper/lti11-response.mapper';
 import { LoggerModule } from '@src/core/logger';
-import { LtiToolService } from '@src/modules/tool/service/lti-tool.service';
 import { SchoolModule } from '@src/modules/school';
-import { ToolController } from '@src/modules/tool/controller/tool.controller';
+import { ExternalToolRepoMapper } from '@shared/repo/externaltool/external-tool.repo.mapper';
+import { ExternalToolRepo } from '@shared/repo/externaltool/external-tool.repo';
+import { AuthorizationModule } from '@src/modules/authorization';
+import { OauthProviderServiceModule } from '@shared/infra/oauth-provider';
+import { EncryptionModule } from '@shared/infra/encryption';
 import { Lti11Uc } from './uc/lti11.uc';
 import { LtiRoleMapper } from './mapper/lti-role.mapper';
 import { Lti11Service } from './service/lti11.service';
 import { UserModule } from '../user';
+import { ToolController } from './controller/tool.controller';
+import { Lti11ResponseMapper } from './mapper/lti11-response.mapper';
+import { ExternalToolUc } from './uc/external-tool.uc';
+import { ExternalToolService } from './service/external-tool.service';
+import { ExternalToolRequestMapper } from './mapper/external-tool-request.mapper';
+import { ExternalToolResponseMapper } from './mapper/external-tool-response.mapper';
 
 @Module({
-	imports: [UserModule, LoggerModule, SchoolModule],
+	imports: [UserModule, LoggerModule, SchoolModule, AuthorizationModule, OauthProviderServiceModule, EncryptionModule],
 	controllers: [ToolController],
-	providers: [Lti11Service, Lti11Uc, LtiRoleMapper, Lti11ResponseMapper, LtiToolRepo, PseudonymsRepo, LtiToolService],
+	providers: [
+		Lti11Service,
+		Lti11Uc,
+		LtiRoleMapper,
+		Lti11ResponseMapper,
+		LtiToolRepo,
+		PseudonymsRepo,
+		ExternalToolRepoMapper,
+		ExternalToolRepo,
+		ExternalToolUc,
+		ExternalToolService,
+		ExternalToolRequestMapper,
+		ExternalToolResponseMapper,
+	],
+	exports: [ExternalToolRepo, LtiToolRepo],
```

Comment:
where are these repos needed? can that go through an exported service?

Reply:
We export them, so they do not have to be provided everywhere. If there is an architectural decision to do it otherwise, we will gladly do that :) (and no, at the time of this writing these repos are not used anywhere else ^^)

Reply:
if they are not used anywhere, please dont export them.
In the future, there should be no reason to need a repo anywhere outside its module. (services/DOs should be used instead)